searx/engines/yggtorrent.py

```diff
@@ -0,0 +1,125 @@
+#  Yggtorrent (Videos, Music, Files)
+#
+# @website     https://www2.yggtorrent.si
+# @provide-api no (nothing found)
+#
+# @using-api   no
+# @results     HTML (using search portal)
+# @stable      no (HTML can change)
+# @parse       url, title, seed, leech, publishedDate, filesize
+
+from lxml import html
+from operator import itemgetter
+from datetime import datetime
+from searx.engines.xpath import extract_text
+from searx.url_utils import quote
+from searx.utils import get_torrent_size
+from searx.poolrequests import get as http_get
+
+# engine dependent config
+categories = ['videos', 'music', 'files']
+paging = True
+
+# search-url
+url = 'https://www2.yggtorrent.si/'
+search_url = url + 'engine/search?name={search_term}&do=search&page={pageno}&category={search_type}'
+
+# yggtorrent specific type-definitions
+search_types = {'files': 'all',
+                'music': '2139',
+                'videos': '2145'}
+
+cookies = dict()
+
+
+def init(engine_settings=None):
+    global cookies
+    # initial cookies
+    resp = http_get(url, allow_redirects=False)
+    if resp.ok:
+        for r in resp.history:
+            cookies.update(r.cookies)
+        cookies.update(resp.cookies)
+
+# do search-request
+
```

Comment:
nitpicking: can you remove the two blank lines ?

Reply:
Unfortunately no because without that pep8 complains about not having enough blank lines:
```
/usr/lib/python3.8/site-packages/pep8.py:110: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
/usr/lib/python3.8/site-packages/pep8.py:2123: UserWarning: 

pep8 has been renamed to pycodestyle (GitHub issue #466)
Use of the pep8 tool will be removed in a future release.
Please install and use `pycodestyle` instead.

$ pip install pycodestyle
$ pycodestyle ...

  warnings.warn(
searx/engines/yggtorrent.py:44:1: E302 expected 2 blank lines, found 0
```

Reply:
```python
        cookies.update(resp.cookies)


# do search-request
def request(query, params):
```

This warning:
```
/home/alexandre/code/searx/local/py3/lib/python3.7/site-packages/pep8.py:109: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
```
remains, but not related: https://github.com/PyCQA/pycodestyle/issues/728